src/overlay/Peer.cpp

```diff
@@ -137,6 +141,39 @@ Peer::MsgCapacityTracker::getPeer()
     return mWeakPeer;
 }
 
+bool
+Peer::isPeerSlow()
+{
+    auto now = mApp.getClock().now();
```

Comment:
I don't think you want to categorize peers like this:
you probably want to categorize peers all at the same time as to "snapshot" their quality (so maybe from the "tick" callback?).